src/v/kafka/topic_probe.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/configuration.h"
+#include "prometheus/prometheus_sanitize.h"
+
+#include <seastar/core/metrics.hh>
+
+namespace kafka {
+class topic_probe {
+public:
+    void setup_metrics() {
+        namespace sm = ss::metrics;
+
+        if (config::shard_local_cfg().disable_metrics()) {
+            return;
+        }
+
+        std::vector<sm::label_instance> labels{sm::label("request")("produce|consume")};
```

Comment:
When I read the metric specification (see below) I interpreted it to mean that the `request` label should take either the `produce` or `consume` value depending on the request being processed. To achieve this you'd need to have two metrics in the `add_group` call: one with the `consume` label and the other with the `produce` label.

```
# HELP redpanda_kafka_request_bytes_total Number of bytes in client payloads by request type
# TYPE redpanda_kafka_request_bytes_total counter
redpanda_kafka_request_bytes_total{request="produce|consume",namespace,topic}
```